src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -0,0 +1,205 @@
+
+    private unsubscribeFrom(subscription: Subscription): void {
+        if (subscription) {
+            subscription.unsubscribe();
+        }
+    }
+
+    private checkGroupMatch(draggableGroup: string|string[]): boolean {
+        if (!draggableGroup && this._group) {
+            return false;
+        }
+
+        if (!this._group && draggableGroup) {
+            return false;
+        }
+
+        if (!this._group && !draggableGroup) {
+            return true;
+        }
+
+        if (typeof draggableGroup === "string") {
+            if (typeof this._group === "string") {
+                return this._group === draggableGroup;
+            } else {
+                return this._group.indexOf(draggableGroup) > -1;
+            }
+        } else {
+            if (typeof this._group === "string") {
+                return draggableGroup.indexOf(this._group) > -1;
+            } else {
+                return (this._group as string[]).filter(groupKey => draggableGroup.indexOf(groupKey) > -1).length > 0;
+            }
+        }
+    }
+
+    private isInDropArea(point: {pageX: number; pageY: number}): boolean {
+        if (!point) {
+            return false;
+        }
+
+        if (!this.clientRect) {
+            this.clientRect = this.domAdapter.clientRect(this.droppableEl);
+        }
+
+        if (point.pageX >= this.clientRect.left - this._dropTolerance.left &&
+            point.pageX <= this.clientRect.right + this._dropTolerance.right &&
+            point.pageY >= this.clientRect.top - this._dropTolerance.top &&
+            point.pageY <= this.clientRect.bottom + this._dropTolerance.bottom) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private onDragStart(dragStartEvent: ClrDragEvent<T>): void {
+        // Check draggable and droppable have a matching group key.
+        this.isDraggableMatch = this.checkGroupMatch(dragStartEvent.group);
+
+        // Subscribe to dragMoved and dragEnded only if draggable and droppable have a matching group key.
+        if (this.isDraggableMatch) {
+            this.dragStartEmitter.emit(dragStartEvent);
+            this.dragMoveSubscription = this.eventBus.dragMoved.subscribe((dragMoveEvent: ClrDragEvent<T>) => {
+                this.onDragMove(dragMoveEvent);
+            });
+            this.dragEndSubscription = this.eventBus.dragEnded.subscribe((dragEndEvent: ClrDragEvent<T>) => {
+                this.onDragEnd(dragEndEvent);
+            });
+        }
+    }
+
+    private onDragMove(dragMoveEvent: ClrDragEvent<T>): void {
+        if (!this._isDraggableOver && this.isInDropArea(dragMoveEvent.dropPointPosition)) {
+            this.isDraggableOver = true;
+            const dragEnterEvent = this.transformAndBroadcastEvent(dragMoveEvent, ClrDragEventType.DRAG_ENTER);
+            this.dragEnterEmitter.emit(dragEnterEvent);
+        } else if (this._isDraggableOver && !this.isInDropArea(dragMoveEvent.dropPointPosition)) {
```

Comment:
Nitpick: you should just call `this.isInDropArea(dragMoveEvent.dropPointPosition)` at the beginning of this method and use the result for both `if`s. Right now you're caching the `clientRect` so the extra computation is almost nothing, but I'm scared we might make it heavier in the future and this is a free optimization.